### PR TITLE
Downgrade exception to warning in case of entity id duplication.

### LINKED ISSUE #1551: Cannot change IDs for Residue
### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

```python
import sys; print(sys.version)
import platform; print(platform.python_implementation()); print(platform.platform())
import Bio; print(Bio.__version__)
```

[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)]
CPython
Darwin-17.3.0-x86_64-i386-64bit
1.70

### Expected behaviour

`Residue.id(value) or res.id = value`
changes the id of the residue. This is used for renumbering a PDB.
This used to work fine for many years and now results in an error, with no obvious override mechanism. 

### Actual behaviour

```
ValueError: Cannot change id from `(' ', 8, ' ')` to `(' ', 9, ' ')`. The id `(' ', 9, ' ')` is already used for a sibling of this entity.
```

In addition, using `detach_child()` on the residue for the 'already in use sibling' does not work and results in this error:

```
  File "/Library/Python/2.7/site-packages/Bio/PDB/Entity.py", line 120, in detach_child
    child = self.child_dict[id]
KeyError: (' ', 9, ' ')
```

### Steps to reproduce

Change the id of a residue to another residue number, that is a 'child'

### PR DESCRIPTION
**Pull request by JoaoRodrigues**
- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit``
locally, and understand that continuous integration checks will be used to
confirm the Biopython unit tests and style checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

This potentially closes #1551. It might open a pandora box of errors if people abuse this or use this without knowing what's going on under the hood.

Closes #...

### CODE REVIEW

File: Bio/PDB/Entity.py [hunk 1]
```diff
@@ -173,9 +175,11 @@ def id(self, value):
...
         if self.parent:
             if value in self.parent.child_dict:
-                raise ValueError(
-                    f"Cannot change id from `{self._id}` to `{value}`."
-                    f" The id `{value}` is already used for a sibling of this entity."
+                # See issue 1551 for details on the downgrade.
+                warnings.warn(
+                    f"The id `{value}` is already used for a sibling of this entity. "
+                    f"Changing id from `{self._id}` to `{value}` might create access "
+                    "inconsistencies to children of the parent entity."
```
Comment (peterjc) [inline Bio/PDB/Entity.py:182]: Use ``from Bio import BiopythonWarning`` to make it slightly easier for people to silence this if they really want to do this?

**Review by peterjc — APPROVED** [review]
Squash-and-merge?

**Comment by mdehoon** [comment]
@JoaoRodrigues  Are you aware that by default warning can get triggered only once in a Python session?

**Comment by JoaoRodrigues** [comment]
Hi Michiel,

Yes - I preferred using a warning vs a print statement for that reason,
otherwise e.g. you'd have the warning printed for every aminoacid in a
protein when you're renumbering them and that's annoying.

A segunda, 26/02/2024, 09:19, mdehoon ***@***.***> escreveu:

**Comment by ALGW71** [comment]
Hi Guys,

Was this actually downgraded to a warning?

I see the merge above, but I am using version 1.83 and still get the following error:

1igt.pdb could not be parsed: Cannot change id from `(' ', 10, ' ')` to `(' ', 11, ' ')`. The id `(' ', 11, ' ')` is already used for a sibling of this entity.

Thanks,
Alex

**Comment by ALGW71** [comment]
Can confirm this works with the dev version of 1.84.

Thanks for solving this!

**Pull request merged**